sklearn/decomposition/sparse_pca.py

```diff
@@ -283,6 +348,17 @@ def fit(self, X, y=None):
         """
         random_state = check_random_state(self.random_state)
         X = check_array(X)
+
+        if self.normalize_components:
+            self.mean_ = np.mean(X, axis=0)
```

Comment:
I would almost call `X.mean(axis=0)`

Reply:
I'm following this but what's the difference ? Except that it is a bit more compact is there a performance improvement ?

Reply:
X.mean() will work for both dense and sparse matrices. But it does not matter since we don't have sparse. So this change is hope to you :)

Reply:
Ok we never know maybe one day so it can't hurt to change this.